[IMP] payment_providers: payment methods + restructure doc page (#7414)
odoo/documentation: content/applications/finance/payment_providers.rst

:show-content:

=====================================

> Payment providers and payment methods

Author: Ok then, I changed the title but will keep the file name as is for now.

  - Review and verify that orders are legitimate before the payment is completed and the fulfillment
    process starts.
  - Avoid potentially high processing fees for cancelled payments: payment providers will not charge
    you for voiding an authorization.
  - Hold a security deposit to return later, minus any deductions (e.g., after a damage).
.. note::
   By default, the payment provider remains **unpublished** in test mode.

Enable this feature by navigating to the :guilabel:`Configuration` tab from your payment provider
and by ticking the :guilabel:`Capture Amount Manually` checkbox.
.. warning::
   We recommend using the test mode on a duplicate or a test database to avoid potential issues
   with your invoice numbering.

To capture the payment after it was authorized, go to the related sales order or invoice and click
on the :guilabel:`CAPTURE TRANSACTION` button. To release the funds, click on the :guilabel:`VOID
TRANSACTION` button.
.. _payment_providers/payment_methods:

If your provider supports partial capture, you can select to capture only a **partial amount** of
the full amount. The **remaining (uncaptured) amount** can either be voided or partially captured

> again later.

Reviewer: Was this dropped on purpose?

> .. warning::

Reviewer: ?
```suggestion
.. danger::
```

   - Each payment method is preconfigured in a way that aligns with the payment providers'
     behavior and their integration with Odoo. Any change to this configuration may result in errors
     and should be tested on a duplicate or test database first.
   - Modifications to the payment method's configuration only work to the extent of the method
     and provider's capabilities. For example, you can disable :ref:`tokenization
     <payment_providers/features/tokenization>` or remove :ref:`supported countries
     <payment_providers/currencies_countries>` for a payment method if you wish to. However, adding
     countries for a method that is only supported in on country or enabling tokenization for a
     method linked to a provider that does not support it will not produce the intended results.

> .. _payment_providers/features/tokenization:

Reviewer: ```suggestion
.. _payment_providers/tokenization:
```

Tokenization
============

:ref:`If the payment provider supports this feature <payment_providers/online_providers>`, customers

> can choose to save their card details for later. To enable this feature, go to the

Reviewer: It's less and less about cards only.
```suggestion
can choose to save their payment method details for later. To enable this feature, go to the
```

payments, without the customer having to enter their card details again. This is particularly useful
for the eCommerce conversion rate and for subscriptions that use recurring payments.

.. note::
   You remain fully PCI-compliant when you enable this feature because Odoo does not store the card
   details directly. Instead, it creates a payment token that only holds a reference to the card
   details stored on the payment provider's server.

.. _payment_providers/features/manual_capture:

Manual capture
==============

:ref:`If the payment provider supports this feature <payment_providers/online_providers>`, you can
authorize and capture payments in two steps instead of one. To enable this feature, go to the
:guilabel:`Configuration` tab of the selected payment provider and enable :guilabel:`Capture Amount
Manually`.

When you authorize a payment, the funds are reserved on the customer's payment method but they are
not immediately charged. The charge is only made when you manually capture the payment later on.
You can also void the authorization to release the reserved funds; this is equivalent to a regular

> cancellation. Capturing payments manually is useful in many situations:

Reviewer: Cancellations are more of a customer action than a vendor action. Vendors typically have to refund payments that they want to revert when they have already been captured. The difference with a void is that a void releases the funds by canceling the authorization before the capture moves any money. In practice, you don't have to pay expensive (several percent) refund fees.

https://support.stripe.com/questions/understanding-fees-for-refunded-payments

Reviewer: See third bullet point.

- Receive the payment confirmation and wait until the order is shipped to capture the payment.
- Review and verify that orders are legitimate before the payment is completed and the fulfillment
  process starts.

> - Avoid potentially high processing fees for cancelled payments: payment providers will not charge

Reviewer: refund fees; refunded payments

To capture the payment after it was authorized, go to the related sales order or invoice and click
the :guilabel:`Capture Transaction` button. To release the funds, click the :guilabel:`Void
Transaction` button.

.. note::
   - Some payment providers support capturing only part of the authorized amount. The remaining
     amount can then be either captured or voided. These providers have the value **Full and
     partial** in the :ref:`table above <payment_providers/online_providers>`. The providers that
     only support capturing or voiding the full amount have the value **Full only**.
   - The funds are likely not reserved forever. After a certain time, they may be automatically
     released back to the customer's payment method. Refer to your payment provider's documentation
     for the exact reservation duration.
   - Odoo does not support this feature for all payment providers but some allow the manual capture
     from their website interface.

.. _payment_providers/features/express_checkout:

Express checkout
================

:ref:`If the payment provider supports this feature <payment_providers/online_providers>`, you can

> allow customers to use the :guilabel:`Google Pay` and :guilabel:`Apple Pay` buttons and pay their

Reviewer: Not sure if it's worth mentioning but there is also Stripe's Link now (quite anecdotic).

eCommerce orders in one click. When they use one of these buttons, customers go straight from the
cart to the confirmation page without having to fill the contact form. They just have to validate
the payment on Google's or Apple's payment form.

> To enable this feature, go to the :guilabel:`Configuration` tab of the selected payment provider and

Reviewer: For Stripe, they should also hit the "Enable Apple Pay" button.

is possible to add or remove a payment provider's supported currencies or countries.

.. note::

   - :ref:`Payment methods <payment_providers/payment_methods>` also have their own list of
     available currencies and countries that serves as a filter during payment operations.
   - If the list of supported currencies or countries is empty, it either means the list is too long
     to be displayed, or Odoo does not have information on that payment provider. The payment
     provider remains available, even though it is possible the payment will be refused at a later
     stage should the currency not be supported.

Maximum amount
--------------

You can restrict the :guilabel:`Maximum Amount` that can be paid with the selected provider. Leave
the field to `0.00` to make the payment provider available regardless of the payment amount.

.. important::
   This feature is not intended to work on pages that allow the customer to update the payment
   amount, e.g., the **Donation** snippet and the **Checkout** page when paid :doc:`shipping methods
   <../websites/ecommerce/checkout_payment_shipping/shipping>` are enabled.

.. _payment_providers/journal:

> Payment journal

Reviewer: Do what you want with this information: payment journals must be configured only if Invoicing/Accounting is installed, since Odoo 17.

.. note::
   - The payment journal must be a :guilabel:`Bank` journal.
   - The same journal can be used for several payment providers.

Accounting perspective
----------------------

From an accounting perspective, there are two types of online payment workflows: the payments that
go directly on your bank account and follow the usual :doc:`reconciliation
<accounting/bank/reconciliation>` workflow and those coming from :ref:`online payment providers
<payment_providers/online_providers>` that are third-party services and require you to follow
another accounting workflow. For these payments, you need to consider how you want to record your
payments' journal entries. We recommend you to ask your accountant for advice.

By default, the :guilabel:`Bank Account` defined for the :ref:`payment journal
<payment_providers/journal>` is used, but you can also define an **Outstanding account** for each
payment provider to separate the provider's payments from other payments in the
:guilabel:`Incoming Payments` and :guilabel:`Outgoing Payments` tabs.

.. image:: payment_providers/bank_journal.png
   :alt: Define an outstanding account for a payment provider.

.. _payment_providers/refunds:

> Refunds

Reviewer: I think you forgot this one; it should be right after manual capture.
